fedimint-client/src/dynamic_config.rs

```diff
@@ -0,0 +1,271 @@
+    /// background task continuously updating the registered config fields. The
+    /// task will shut down when the supplied task manager shuts down.
+    pub fn start(self, task_group: &TaskGroup, api: &impl AnyDynApi) -> DynConfigService {
+        let db = self
+            .db
+            .expect("Database must be set before starting the service");
+        let db_key_prefix = self
+            .db_key_prefix
+            .expect("DB key prefix must be set before starting the service");
+
+        let updatet_tasks = self
+            .tracked_values
+            .iter()
+            .map(|tracked_value| {
+                let db = db.clone();
+                let api = api.clone();
+                let fetch_method = tracked_value.fetch_method;
+                let update_interval = tracked_value.update_interval;
+
+                async move {
+                    loop {
+                        if let Err(e) =
+                            update_config_value(&db, db_key_prefix, &api, fetch_method).await
+                        {
+                            warn!(%fetch_method, ?e, "Updating dyn config field failed");
+                        };
+                        sleep(update_interval).await;
+                    }
+                }
+            })
+            .collect::<Vec<_>>();
+        task_group.spawn_cancellable("dyn_cfg_sync", async move {
+            join_all(updatet_tasks).await;
+        });
+
+        DynConfigService {
+            db_key_prefix,
+            tracked_values: self
+                .tracked_values
+                .into_iter()
+                .map(|tv| tv.fetch_method)
+                .collect(),
+        }
+    }
+}
+
+async fn update_config_value(
+    db: &Database,
+    db_key_prefix: u8,
+    api: &impl AnyDynApi,
+    fetch_method: &'static str,
+) -> anyhow::Result<()> {
+    let fetched_value = api
+        .request_current_consensus::<serde_json::Value>(
+            fetch_method.to_owned(),
+            ApiRequestErased::default(),
+        )
+        .await?;
+
+    let db_key = DynConfigKey {
+        db_key_prefix,
+        fetch_method: fetch_method.to_owned(),
+    };
+    let db_value = DynConfigValue {
+        value: fetched_value,
+        last_update: now(),
+    };
+
+    db.autocommit(
+        |dbtx, _pd| {
+            let key_bytes = db_key.consensus_encode_to_vec();
+            let value_bytes = db_value.consensus_encode_to_vec();
+            Box::pin(async move { dbtx.raw_insert_bytes(&key_bytes, &value_bytes).await })
+        },
+        None,
+    )
+    .await
+    .expect("DB operation cannot fail");
+
+    Ok(())
+}
+
+/// Service that keeps track of a fixed list of dynamic config fields. These are
+/// regularly re-fetched from the server and saved in the DB. Use
+/// `[DynConfigService::get_config_value]` to retrieve the latest value.
+#[derive(Debug)]
+pub struct DynConfigService {
+    db_key_prefix: u8,
```

Comment:
Thought: maybe modules should just get this functionality on the their `ClientModuleInitArgs` and provide their own `Key`, etc. to avoid having to deal with module specific stuff in the general code.

Each time we do module-stuff outside of it it turns very complex, requires Dyn-erased types, etc.